tests/test_add.py

```diff
@@ -230,3 +282,123 @@ def test_without_exclusions_ok(
+
+        assert otu_after.excluded_accessions == set(original_accessions)
+
+    @pytest.mark.parametrize(
+        "taxid, original_accessions, refseq_accessions",
+        [
+            (1169032, ["AB017503"], ["NC_003355"]),
+            (345184, ["DQ178608", "DQ178609"], ["NC_038792", "NC_038793"])
+        ]
+    )
+    def test_automatic_replace_ok(
+        self,
+        empty_repo,
+        taxid: int,
+        original_accessions: list[str],
+        refseq_accessions: list[str],
+    ):
+        """Test that an attempt to add an isolate with the same name as a previous isolate
+        maintains the same isolate UUID, but replaces the existing sequences including
+        sequence UUIDs."""
+        otu_before = create_otu(empty_repo, taxid, accessions=original_accessions, acronym="")
+
+        otu_before = empty_repo.get_otu(otu_before.id)
+
+        assert otu_before.accessions == set(original_accessions)
+
+        updated_isolate = add_isolate(empty_repo, otu_before, refseq_accessions)
+
+        assert updated_isolate is not None
+
+        otu_after = empty_repo.get_otu(otu_before.id)
+
+        assert otu_after.isolate_ids == otu_before.isolate_ids
+
+        assert otu_after.accessions == set(refseq_accessions)
+
+        assert otu_after.excluded_accessions == set(original_accessions)
+
+        assert otu_after.sequence_ids != otu_before.sequence_ids
+
+
+class TestRemoveIsolate:
+    @pytest.mark.parametrize(
+        "taxid, isolate_name",
+        [
+            (1169032, IsolateName(type=IsolateNameType.ISOLATE, value="WMoV-6.3"))
+        ]
+    )
+    def test_ok(self, scratch_repo, taxid: int, isolate_name: IsolateName):
+        """Test that a given isolate can be removed from the OTU."""
+        otu_before = scratch_repo.get_otu_by_taxid(taxid)
+
+        isolate_id = otu_before.get_isolate_id_by_name(isolate_name)
+
+        assert type(isolate_id) is UUID
+
+        remove_isolate_from_otu(scratch_repo, otu_before, isolate_id)
+
+        otu_after = scratch_repo.get_otu_by_taxid(taxid)
+
+        assert otu_after.get_isolate(isolate_id) is None
+
+        assert otu_before.get_isolate(isolate_id).accessions not in otu_after.accessions
+
+        assert len(otu_after.isolate_ids) == len(otu_before.isolate_ids) - 1
+
+
+class TestReplaceSequence:
+    def test_ok(self, precached_repo):
+        otu_before = create_otu(
+            precached_repo,
+            1169032,
+            ["MK431779"],
+            acronym=""
+        )
+        """Test that a sequence in an OTU can be replaced manually."""
+        isolate_id, old_sequence_id = otu_before.get_sequence_id_hierarchy_from_accession(
+            "MK431779"
+        )
+
+        assert type(old_sequence_id) is UUID
+
+        sequence = replace_sequence_in_otu(
+            repo=precached_repo,
+            otu=otu_before,
+            new_accession="NC_003355",
+            replaced_accession="MK431779"
+        )
+
+        assert sequence is not None
```

Comment:
You should assert on at least a few properties to make sure it is returning the correct sequence object.